unittests/unittest_radix_tree.cpp

```diff
@@ -0,0 +1,324 @@
+    TEST_ASSERT_FALSE (tree_rm (tree, "check"));
+}
+
+void test_rm_common_prefix_entry ()
+{
+    zmq::radix_tree tree;
+
+    tree_add (tree, "checkpoint");
+    tree_add (tree, "checklist");
+    tree_add (tree, "check");
+    TEST_ASSERT_TRUE (tree_rm (tree, "check"));
+}
+
+void test_rm_null_entry ()
+{
+    zmq::radix_tree tree;
+
+    tree_add (tree, "");
+    TEST_ASSERT_TRUE (tree_rm (tree, ""));
+}
+
+void test_check_empty ()
+{
+    zmq::radix_tree tree;
+
+    TEST_ASSERT_FALSE (tree_check (tree, "foo"));
+}
+
+void test_check_added_entry ()
+{
+    zmq::radix_tree tree;
+
+    tree_add (tree, "entry");
+    TEST_ASSERT_TRUE (tree_check (tree, "entry"));
+}
+
+void test_check_common_prefix ()
+{
+    zmq::radix_tree tree;
+
+    tree_add (tree, "introduce");
+    tree_add (tree, "introspect");
+    TEST_ASSERT_FALSE (tree_check (tree, "intro"));
+}
+
+void test_check_prefix ()
+{
+    zmq::radix_tree tree;
+
+    tree_add (tree, "toasted");
+    TEST_ASSERT_FALSE (tree_check (tree, "toast"));
+    TEST_ASSERT_FALSE (tree_check (tree, "toaste"));
+    TEST_ASSERT_FALSE (tree_check (tree, "toaster"));
+}
+
+void test_check_nonexistent_entry ()
+{
+    zmq::radix_tree tree;
+
+    tree_add (tree, "red");
+    TEST_ASSERT_FALSE (tree_check (tree, "blue"));
+}
+
+void test_check_query_longer_than_entry ()
+{
+    zmq::radix_tree tree;
+
+    tree_add (tree, "foo");
+    TEST_ASSERT_TRUE (tree_check (tree, "foobar"));
+}
+
+void test_check_null_entry_added ()
+{
+    zmq::radix_tree tree;
+
+    tree_add (tree, "");
+    TEST_ASSERT_TRUE (tree_check (tree, "all queries return true"));
+}
+
+void test_size ()
+{
+    zmq::radix_tree tree;
+
+    // Adapted from the example on wikipedia.
+    std::vector<std::string> keys;
+    keys.push_back ("tester");
+    keys.push_back ("water");
+    keys.push_back ("slow");
+    keys.push_back ("slower");
+    keys.push_back ("test");
+    keys.push_back ("team");
+    keys.push_back ("toast");
+
+    for (size_t i = 0; i < keys.size (); ++i)
+        TEST_ASSERT_TRUE (tree_add (tree, keys[i]));
+    TEST_ASSERT_TRUE (tree.size () == keys.size ());
+    for (size_t i = 0; i < keys.size (); ++i)
+        TEST_ASSERT_FALSE (tree_add (tree, keys[i]));
+    TEST_ASSERT_TRUE (tree.size () == 2 * keys.size ());
+    for (size_t i = 0; i < keys.size (); ++i)
+        TEST_ASSERT_FALSE (tree_rm (tree, keys[i]));
+    TEST_ASSERT_TRUE (tree.size () == keys.size ());
+    for (size_t i = 0; i < keys.size (); ++i)
+        TEST_ASSERT_TRUE (tree_rm (tree, keys[i]));
+    TEST_ASSERT_TRUE (tree.size () == 0);
+}
+
+void return_key (unsigned char *data, size_t size, void *arg)
+{
+    std::vector<std::string> *vec =
+      reinterpret_cast<std::vector<std::string> *> (arg);
+    std::string key;
+    for (size_t i = 0; i < size; ++i)
+        key.push_back (static_cast<char> (data[i]));
+    vec->push_back (key);
+}
+
+void test_apply ()
+{
+    zmq::radix_tree tree;
+
+    std::unordered_set<std::string> keys;
```

Comment:
This is only available in C++11. But I think this can simply be changed to std::set

Reply:
Thanks, fixed!